### PR TITLE
Use ConfigFactory to create the default config instances.

### PR DESCRIPTION
**Pull request by gurbuzali**
Currently, we use default constructor to create Config, ClientConfig
and ClientFailoverConfig when parsing the spring application-context.

With these changes, we introduce a factory class to create these config
instances. The factory class uses 'Supplier's to create the config
instances. These suppliers can be changed to provide a pre-configured
instance instead of the default one.

The motivation to use factory methods is that some of the tests stuck
at creating the client, the default connection timeout for the client
is infinite. Instead of changing each and every config xml file, we
want to pre-configure the client (and member for smallInstancesConfig)
and add a finite value for the connection timeout in the tests.

Checklist:
- [x] Labels and Milestone set
- [ ] Added a line in `hazelcast-jet-distribution/src/root/release_notes.txt` (for any non-trivial fix/enhancement/feature)
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

**Comment by gurbuzali** [comment]
both of the failed tests are unrelated, can be merged after review, cc @frant-hartm

**Review by frant-hartm — APPROVED** [review]

**Comment by frant-hartm** [comment]
I see nothing wrong there, it feels a bit uneasy to change bean definition parsers for tests, but I don't have a better solution.

Would be great to get another reviewer.

**Review by sancar — APPROVED** [review]

**Pull request merged**
